behandlingsprosess/src/main/java/no/nav/ung/sak/domene/behandling/steg/varselrevurdering/VarselRevurderingStegImpl.java

```diff
@@ -76,4 +106,85 @@ private List<UngdomsprogramBekreftetPeriodeEndring> finnBekreftelser(Behandling
             .flatMap(it -> it.getBekreftetPeriodeEndringer().stream()).toList();
     }
 
+    private void opprettTaskForEtterlysning(BehandlingskontrollKontekst kontekst, LocalDateTimeline<Boolean> endretUngdomsprogramTidslinje) {
+        UngdomsprogramPeriodeGrunnlag ungdomsprogramPeriodeGrunnlag = ungdomsprogramPeriodeRepository.hentGrunnlag(kontekst.getBehandlingId()).orElseThrow();
+
+        List<Trigger> triggere = prosessTriggereRepository.hentGrunnlag(kontekst.getBehandlingId())
+            .stream()
+            .map(ProsessTriggere::getTriggere)
+            .flatMap(Collection::stream)
+            .toList();
+
+        LocalDateTimeline<Set<Etterlysning>> etterlysningerLocalDateTimeline = triggere
+            .stream()
+            .map(trigger -> {
+                    Etterlysning etterlysning = mapTilEtterlysning(kontekst, trigger, ungdomsprogramPeriodeGrunnlag);
+                    if (etterlysning == null) {
+                        return null;
+                    }
+                    return new LocalDateTimeline(
+                        trigger.getPeriode().getFomDato(),
+                        trigger.getPeriode().getTomDato(),
+                        Set.of(etterlysning));
+                }
+            )
+            .filter(Objects::nonNull)
+            .reduce((t1, t2) -> t1.crossJoin(t2, StandardCombinators::union))
+            .orElse(LocalDateTimeline.empty());
+
+        etterlysningerLocalDateTimeline.stream()
+            .filter(Objects::nonNull)
+            .forEach(etterlysningSegment -> {
+                Etterlysning etterlysning = etterlysningSegment.getValue().iterator().next();
+                List<Etterlysning> eksiterendeEtterLysninger = etterlysningRepository.hentEtterlysninger(kontekst.getBehandlingId(), etterlysning.getType());
```

Comment:
Kanskje vi burde trekke dette ut fra foreachen og så heller lage ei liste med etterlysninger som skal opprettes og etterlysninger som skal avbrytes basert på kva etterlysninger som finnes og kva som er gjeldende startdato/sluttdato for programmet

Reply:
Eg tenkjer at vi skal kun opprette ein etterlysning av kvar type. Denne etterlysningen kan vi finne ved å f.eks sortere triggere på opprettet tidspunkt evt sjekke kva som er gjeldende programperiode